dotnet/src/SemanticKernel.Skills/Skills.Memory.Qdrant/QdrantVectorDbClient.cs

```diff
@@ -0,0 +1,512 @@
+/// An implementation of a client for the Qdrant VectorDB. This class is used to
+/// connect, create, delete, and get embeddings data from a Qdrant VectorDB instance.
+/// </summary>
+public class QdrantVectorDbClient<TEmbedding>
+    where TEmbedding : unmanaged
+{
+    /// <summary>
+    /// The endpoint for the Qdrant service.
+    /// </summary>
+    public string BaseAddress
+    {
+        get { return this._httpClient.BaseAddress.ToString(); }
+    }
+
+    /// <summary>
+    /// The port for the Qdrant service.
+    /// </summary>
+    public int Port
+    {
+        get { return this._httpClient.BaseAddress.Port; }
+    }
+
+    /// <summary>
+    /// The constructor for the QdrantVectorDbClient.
+    /// </summary>
+    /// <param name="endpoint"></param>
+    /// <param name="port"></param>
+    /// <param name="httpClient"></param>
+    /// <param name="log"></param>
+    public QdrantVectorDbClient(
+        string endpoint,
+        int? port = null,
+        HttpClient? httpClient = null,
+        ILogger? log = null)
+    {
+        this._log = log ?? NullLogger<QdrantVectorDbClient<TEmbedding>>.Instance;
+
+        Verify.ArgNotNullOrEmpty(endpoint, nameof(endpoint));
+
+        this._httpClient = httpClient ?? new HttpClient(HttpHandlers.CheckCertificateRevocation);
+        this._httpClient.BaseAddress = SanitizeEndpoint(endpoint, port);
+    }
+
+    /// <summary>
+    /// Get a specific vector by its unique Qdrant ID.
+    /// </summary>
+    /// <param name="collectionName"></param>
+    /// <param name="pointId"></param>
+    /// <returns></returns>
+    public async Task<DataEntry<QdrantVectorRecord<TEmbedding>>?> GetVectorByIdAsync(string collectionName, string pointId)
+    {
+        this._log.LogDebug("Searching vector by point ID");
+
+        using HttpRequestMessage request = GetVectorsRequest.Create(collectionName)
+            .WithPointId(pointId)
+            .WithPayloads(true)
+            .WithVectors(true)
+            .Build();
+
+        (HttpResponseMessage response, string responseContent) = await this.ExecuteHttpRequestAsync(request);
+        try
+        {
+            response.EnsureSuccessStatusCode();
+        }
+        catch (HttpRequestException e)
+        {
+            this._log.LogDebug("Vector not found {0}", e.Message);
+            return null;
+        }
+
+        var data = JsonSerializer.Deserialize<GetVectorsResponse<TEmbedding>>(responseContent);
+
+        if (data == null)
+        {
+            this._log.LogWarning("Unable to deserialize Get response");
+            return null;
+        }
+
+        if (!data.Result.Any())
+        {
+            this._log.LogWarning("Vector not found");
+            return null;
+        }
+
+        var recordData = data.Result.First();
+
+#pragma warning disable CS8604 // The request specifically asked for a payload to be in the response
+        var record = new QdrantVectorRecord<TEmbedding>(
+            new Embedding<TEmbedding>(
```

Comment:
Embedding should be replaced by float[] in QdrantVectorDbClient and its dependencies to @dluc's point above.

Reply:
See my other comment addressed to @dluc